server/schedule/config/config.go

```diff
@@ -0,0 +1,95 @@
+package config
+
+import (
+	"sync"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/core/storelimit"
+)
+
+// RejectLeader is the label property type that suggests a store should not
+// have any region leaders.
+const RejectLeader = "reject-leader"
+
+var schedulerMap sync.Map
+
+// RegisterScheduler registers the scheduler type.
+func RegisterScheduler(typ string) {
+	schedulerMap.Store(typ, struct{}{})
+}
+
+// IsSchedulerRegistered checks if the named scheduler type is registered.
+func IsSchedulerRegistered(name string) bool {
+	_, ok := schedulerMap.Load(name)
+	return ok
+}
+
+// Config is the interface that wraps the Config related methods.
+type Config interface {
```

Comment:
I feel this part is hard to maintain and is error-prone in the future. Just wondering if we can reuse ScheduleConfig in server/config/config.go and move it to server/scheduler to remove dependency. It seems that we want to remove the dependency (on server/config) from server/scheduler, but I'm sure if we can let server/config depend on  server/scheduler/config. I mightn't fully understand our work here. I'll sync with you two after holiday. @rleungx @nolouch ^